### ISSUE TITLE
Misleading 1.10.0 release notes

### ISSUE THREAD

**Issue by snowp** [grpc/grpc-java#4225]
The 1.10.0 change list https://github.com/grpc/grpc-java/releases/tag/v1.10.0 says that https://github.com/grpc/grpc-java/pull/4027 is included, but that doesn't actually seem to be the case. The only released version with this fix applied is 1.9.1 through a back port. 

The commit seems to have been merged to master shortly after 1.10.0 was cut:

```
2c8bc71f core: Resolve address of GRPC_PROXY_EXP hostname // commit with fix
f1bc4493 netty: Add test to verify error for TLS failure
c56114ff netty/shaded: Bump shadow plugin version to 2.0.2
dc95465f core: retry part 5, add RetryPolicy data object
722d6f0b okhttp: support Conscrypt security provider
141a1d23 core: install the binary logging client interceptor (#3937)
2601d544 RELEASING.md: include golden files in branching instructions
2f05c230 Start 1.11.0 development cycle
d792a72e Revert "core: make ManagedChannel honor Service config LB " // last commit included in 1.10.0
fbdc217b Revert "core: handle long dns txt records properly, parse service config, and add tests"
```

Would be great to update the release notes to reflect this. Would also love to get a 1.10.1 release with this fix included, but if that's too much work we can probably sit on 1.9.1 until 1.11.0 is out.

**Comment by ericgribkoff** [comment]
Thanks for pointing this out. It was intended to include #4027 in the 1.10.0 release, but I mishandled the branch cut. I've corrected the release notes, and I think we will likely get a 1.10.1 out next week that does actually include #4027. 

cc @ejona86

**Comment by ejona86** [comment]
This was fixed by #4178. We still need to do the v1.10.1 release.

**Comment by ericgribkoff** [comment]
We released v1.10.1 to Maven today. It should show up on Maven shortly; I'll send out the release notes later today. (Note: the next release, 1.11.0, is scheduled for Monday)
